docs/actions-reusable.md

```diff
@@ -0,0 +1,63 @@
+Like [scripted actions](actions-scripts.md), reusable actions are logical units of analytic code.
+However, whereas the source code for a scripted action must be copied from study to study, the source code for a reusable action is shared by different studies.
+
+## Running reusable actions
+
+You can browse existing reusable actions at <https://actions.opensafely.org>.
+Although each is different, they have a common API.
+Consider the following extract from a study's *project.yaml*:
+
+```yaml
+actions:
+  generate_my_cohort:
+    # ...
+
+  run_a_reusable_action:
+    # The reusable action is called "a_reusable_action".
+    # We will run version "v1.0.0".
+    run: a_reusable_action:v1.0.0
+    needs: [generate_my_cohort]
+    outputs:
+      moderately_sensitive:
+        output: output/output_from_a_reusable_action.csv
+```
+
+## Developing reusable actions
+
+A reusable action is a public repo within the [`opensafely-actions`](https://github.com/opensafely-actions) organisation.
+It has a `main` branch, which is the release branch;
+versions are marked with [tags](http://git-scm.com/book/en/v2/Git-Basics-Tagging).
+In the above extract from a study's *project.yaml*, for example, we will run version `v1.0.0` of the reusable action called `a_reusable_action`.
+
+The repo has the following minimal structure:
+
+```sh
+.
+├── README.md
+└── action.yaml
+```
+
+*README.md* contains information about the reusable action, which is displayed by <https://actions.opensafely.org>.
+
+*action.yaml* contains a run command, which is composed of a runtime image (either `python`, `r`, or `stata-mp`) and an entry point.
+The entry point is a path to a script.
+For example:
+
+```yaml
+run: python:latest python action/cli.py
+```
+
+Where *action/cli.py* is:
+
+```python
+def main():
+  print("A reusable action")
+
+if __name__ == "__main__":
+  main()
+```
+
+When developing a reusable action, just as when developing a scripted action, the action's dependencies are made available by the runtime; they are not made available by the action.
+In practice, this means that a Python action's *requirements.txt* is ignored.
```

Comment:
How does an action author know which dependencies are available?

Reply:
I've described where to find dependencies.